languages/go/oso.go

```diff
@@ -142,6 +142,7 @@ func (o Oso) IsAllowed(actor interface{}, action interface{}, resource interface
 	if err != nil {
 		return false, err
 	} else if results != nil {
+		defer query.Cleanup()
```

Comment:
It's very non-obvious why this is needed. Can we add a comment here and to the `Next` method stating that if you don't exhaust the iterator, you need to call `Cleanup` manually?